server/src/main/java/org/elasticsearch/cluster/metadata/MappingMetadata.java

```diff
@@ -56,7 +60,9 @@ public MappingMetadata(CompressedXContent mapping) {
             throw new IllegalStateException("Can't derive type from mapping, no root type: " + mapping.string());
         }
         this.type = mappingMap.keySet().iterator().next();
-        this.routingRequired = routingRequired((Map<String, Object>) mappingMap.get(this.type));
+        Map<String, Object> withoutType = (Map<String, Object>) mappingMap.get(this.type);
+        this.routingRequired = routingRequired(withoutType);
+        this.fieldsForModels = null;
```

Comment:
How does that work on nodes that don't load the mapping? From what I understand the fields for models will be null and we'll miss the inference call?

Reply:
Fair point. I'm expecting the IndexMetadata to have that field when the index is created originally (and thus passed along to the rest of the cluster), and then serialized to disk using the toXContent method. 

What would be the simplest way to test this?